### DISCUSSION TITLE
Determine if database is up

### DISCUSSION THREAD

**Discussion by pmulgaonkar** [gvenzl/oci-oracle-xe#179] [Q&A] [answered]
Just started using this package under Docker running on AlmaLinux 8. Have reached the point where the DB container is fired off, local user created etc.  But I have one question.

When the container is started (initially) or after a stop and start, it seems to take (on my setup) about 2-3 mins before sqlplus from the host is able to connect to XEPDB1. Is there a docker command that will easily tell me when the DB is up and the listener is ready to accept connections?

--prasanna

**Comment by gvenzl** [comment, marked as answer]
Hey @pmulgaonkar,

Thanks a lot for using these images!

The container provides a [`healthcheck.sh`](https://github.com/gvenzl/oci-oracle-xe/blob/main/healthcheck.sh) script that will tell you the state of the database. Depending on your requirements, you can invoke it from the shell or other script to see whether the database is up or not. If the database instance is up, it will return `0`, otherwise `1`.

For example:

```
[gvenzl@localhost oci-oracle-xe]$ podman run -d --name test -e ORACLE_RANDOM_PASSWORD=y gvenzl/oracle-xe:latest
78f0ee7bb9bc939ccabd1a2e68501d0dc70a723fd1623ffd6f9da149ec284514
[gvenzl@localhost oci-oracle-xe]$
[gvenzl@localhost oci-oracle-xe]$
[gvenzl@localhost oci-oracle-xe]$
[gvenzl@localhost oci-oracle-xe]$ podman exec test healthcheck.sh && echo UP || echo NOT READY
NOT READY
[gvenzl@localhost oci-oracle-xe]$ podman exec test healthcheck.sh && echo UP || echo NOT READY
NOT READY
[gvenzl@localhost oci-oracle-xe]$ podman exec test healthcheck.sh && echo UP || echo NOT READY
UP
```
